apps/clock/js/alarm.js

```diff
@@ -156,6 +156,48 @@ define(function(require, exports, module) {
       // Only update the application if this alarm was actually saved
       // (i.e. it has an ID).
       if (this.id) {
+        var self = this,
```

Comment:
Please, stop using this...you can check what I've said at

https://github.com/mozilla-b2g/gaia/blob/master/apps/system/lockscreen/js/lockscreen_inputpad.js#L63-L66

It's an example to use bind instead of the old trick of refering `this`.

Or, if you're willing to learn more, and which I feel it's more proper, you can take a look at ArrowFunction:

https://github.com/mozilla-b2g/gaia/blob/master/apps/system/lockscreen/js/lockscreen_inputpad.js#L77-L81

The MDN links are:

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_objects/Function/bind
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions